.github/workflows/rust.yml

```diff
@@ -34,6 +34,26 @@ jobs:
       - name: Format
         run: ./scripts/tests/format.sh --check
 
+  copyright:
+    name: Copyright Notices
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+      - name: Install check-license and dependencies
+        run: |
+          pip install scripts/check-license
+          pip install -r scripts/check-license/requirements.txt
+      - name: Query files changed
+        id: files_changed
+        uses: Ana06/get-changed-files@v1.2
+        with:
+          filter: '*.rs$'
+      - name: Check copyright notices
+        run: check-license ${{ steps.files_changed.outputs.added_modified }}
+
```

Comment:
The script does offer the option to correct those errors, what's your position to automatically adding a commit with the updated license headers and optionally passing the workflow log in a comment in the PR?
- Add a commit: https://github.com/zeitgeistpm/zeitgeist/actions/runs/985953773/workflow#L138-L147
- Comment success/failure + logs: https://github.com/zeitgeistpm/zeitgeist/actions/runs/985953773/workflow#L173-L189

Reply:
Writing would just trigger in the case of a PR that should be merged into main.

Reply:
And the commit is triggered by placing a label?

Reply:
I'm okay with automating the process, but I have to caution everyone to double-check the diff everytime they do this. And just to prove my point, I botched the test for the `File.write` function because I forgot how `mock` works...

Reply:
Maybe just a quick argument against automating the write: It probably takes just as much time to execute the write locally and then push the changes manually as it takes to trigger the write remotely, but its much safer to make the changes locally and run `git diff` once just to check if there's been a mistake.